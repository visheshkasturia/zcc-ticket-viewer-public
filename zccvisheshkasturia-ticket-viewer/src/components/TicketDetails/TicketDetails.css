.ticketDetailsContainer{
  display: grid;
  grid-template-columns: 50% 50%;
  grid-template-rows: 10% 10% 10% 10% 10% 50%;
  border: 1px solid black;
  height: 550px; 
  width: 500px;
}

.id-details{
  grid-column: 1/ span 1;
  grid-row: 1/ span 1;
  border: 1px solid black;
}

.req-details{
  grid-column: 2/ span 1;
  grid-row: 1/ span 1;
  border: 1px solid black;
}

.assignee-details{
  grid-column: 1/ span 1;
  grid-row: 2/ span 1;
  border: 1px solid black;
}

.org-details{
  grid-column: 2/ span 1;
  grid-row: 2/ span 1;
  border: 1px solid black;
}

.created-details{
  grid-column: 1/ span 1;
  grid-row: 3/ span 1;
  border: 1px solid black;
}

.updated-details{
  grid-column: 2/ span 1;
  grid-row: 3/ span 1;
  border: 1px solid black;
}

.status-details{
  grid-column: 1/ span 1;
  grid-row: 4/ span 1;
  border: 1px solid black;
}

.priority-details{
  grid-column: 2/ span 1;
  grid-row: 4/ span 1;
  border: 1px solid black;
}

.subject-details{
  grid-column: 1/ span 2;
  grid-row: 5/ span 1;
  border: 1px solid black;
}

.desc-details{
  grid-column: 1/ span 2;
  grid-row: 6/ span 1;
  border: 1px solid black;
}

