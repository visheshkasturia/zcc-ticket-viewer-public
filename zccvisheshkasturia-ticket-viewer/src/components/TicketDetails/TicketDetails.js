import React from 'react';
import './TicketDetails.css';

function TicketDetails({ ticket }) {
  return (
    <div className="ticketDetailsContainer">
      <div data-testid="details-id" className="id-details"><b>Ticket ID :</b> {ticket.id}</div>
      <div data-testid="details-reqid" className="req-details"><b>Requester ID :</b> {ticket.requester_id}</div>
      <div data-testid="details-assignid" className="assignee-details"><b>Assignee ID :</b> {ticket.assignee_id}</div>
      <div data-testid="details-orgid" className="org-details"><b>Organization ID :</b> {ticket.organization_id}</div>
      <div data-testid="details-createdat" className="created-details"><b>Date Created :</b> {ticket.created_at}</div>
      <div data-testid="details-updatedat" className="updated-details"><b>Date Updated :</b> {ticket.updated_at}</div>
      <div data-testid="details-status" className="status-details"><b>Ticket Status :</b> {ticket.status}</div>
      <div data-testid="details-priority" className="priority-details"><b>Ticket Priority :</b> {ticket.priority}</div>
      <div data-testid="details-subject" className="subject-details"><b>Subject :</b> {ticket.subject}</div>
      <div data-testid="details-desc" className="desc-details"><b>Description :</b> {ticket.description}</div>
    </div>
  );
}

export default TicketDetails;
