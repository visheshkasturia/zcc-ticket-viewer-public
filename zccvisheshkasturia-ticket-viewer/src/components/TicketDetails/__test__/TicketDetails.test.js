import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import TicketDetails from '../TicketDetails';

afterEach(cleanup);
describe('TicketDetails renders correctly', () => {
  const ticketData = {
    id: 1,
    status: 'open',
    priority: 'urgent',
    subject: 'test subject',
    description: 'test description',
    requester_id: 'ABC123',
    assignee_id: 'XYZ456',
    organization_id: 'ORG789',
    created_at: '2021-12-01T17:53:53Z',
    updated_at: '2021-12-03T18:42:42Z',
  };

  it('id is correct', () => {
    const { getByTestId } = render(<TicketDetails ticket={ticketData} />);
    expect(getByTestId('details-id')).toHaveTextContent('1');
  });

  it('status is correct', () => {
    const { getByTestId } = render(<TicketDetails ticket={ticketData} />);
    expect(getByTestId('details-status')).toHaveTextContent('open');
  });

  it('priority is correct', () => {
    const { getByTestId } = render(<TicketDetails ticket={ticketData} />);
    expect(getByTestId('details-priority')).toHaveTextContent('urgent');
  });

  it('subject is correct', () => {
    const { getByTestId } = render(<TicketDetails ticket={ticketData} />);
    expect(getByTestId('details-subject')).toHaveTextContent('test subject');
  });

  it('desc is correct', () => {
    const { getByTestId } = render(<TicketDetails ticket={ticketData} />);
    expect(getByTestId('details-desc')).toHaveTextContent('test description');
  });

  it('created date is correct', () => {
    const { getByTestId } = render(<TicketDetails ticket={ticketData} />);
    expect(getByTestId('details-createdat')).toHaveTextContent('2021-12-01T17:53:53Z');
  });

  it('updated date is correct', () => {
    const { getByTestId } = render(<TicketDetails ticket={ticketData} />);
    expect(getByTestId('details-updatedat')).toHaveTextContent('2021-12-03T18:42:42Z');
  });

  it('assignee id is correct', () => {
    const { getByTestId } = render(<TicketDetails ticket={ticketData} />);
    expect(getByTestId('details-assignid')).toHaveTextContent('XYZ456');
  });

  it('requester id is correct', () => {
    const { getByTestId } = render(<TicketDetails ticket={ticketData} />);
    expect(getByTestId('details-reqid')).toHaveTextContent('ABC123');
  });

  it('org id is correct', () => {
    const { getByTestId } = render(<TicketDetails ticket={ticketData} />);
    expect(getByTestId('details-orgid')).toHaveTextContent('ORG789');
  });
  it('Matches snapshot', () => {
    const tree = renderer.create(<TicketDetails ticket={ticketData} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
