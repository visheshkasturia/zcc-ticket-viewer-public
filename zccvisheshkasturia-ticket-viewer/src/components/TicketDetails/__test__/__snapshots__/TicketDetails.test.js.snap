// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TicketDetails renders correctly Matches snapshot 1`] = `
<div
  className="ticketDetailsContainer"
>
  <div
    className="id-details"
    data-testid="details-id"
  >
    <b>
      Ticket ID :
    </b>
     
    1
  </div>
  <div
    className="req-details"
    data-testid="details-reqid"
  >
    <b>
      Requester ID :
    </b>
     
    ABC123
  </div>
  <div
    className="assignee-details"
    data-testid="details-assignid"
  >
    <b>
      Assignee ID :
    </b>
     
    XYZ456
  </div>
  <div
    className="org-details"
    data-testid="details-orgid"
  >
    <b>
      Organization ID :
    </b>
     
    ORG789
  </div>
  <div
    className="created-details"
    data-testid="details-createdat"
  >
    <b>
      Date Created :
    </b>
     
    2021-12-01T17:53:53Z
  </div>
  <div
    className="updated-details"
    data-testid="details-updatedat"
  >
    <b>
      Date Updated :
    </b>
     
    2021-12-03T18:42:42Z
  </div>
  <div
    className="status-details"
    data-testid="details-status"
  >
    <b>
      Ticket Status :
    </b>
     
    open
  </div>
  <div
    className="priority-details"
    data-testid="details-priority"
  >
    <b>
      Ticket Priority :
    </b>
     
    urgent
  </div>
  <div
    className="subject-details"
    data-testid="details-subject"
  >
    <b>
      Subject :
    </b>
     
    test subject
  </div>
  <div
    className="desc-details"
    data-testid="details-desc"
  >
    <b>
      Description :
    </b>
     
    test description
  </div>
</div>
`;
